### ISSUE TITLE
InvocationTargetException

### ISSUE THREAD

**Issue by ihor-sviziev** [enriquepiatti/Magicento#129]
After Magicento upgrade sometimes I got next exception:

```
null
java.lang.reflect.InvocationTargetException
    at sun.reflect.GeneratedMethodAccessor1140.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:483)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.a(FileDocumentManagerImpl.java:149)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.access$000(FileDocumentManagerImpl.java:87)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl$1.invoke(FileDocumentManagerImpl.java:116)
    at com.sun.proxy.$Proxy21.beforeDocumentSaving(Unknown Source)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl$4.run(FileDocumentManagerImpl.java:412)
    at com.intellij.pom.core.impl.PomModelImpl.guardPsiModificationsIn(PomModelImpl.java:354)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.a(FileDocumentManagerImpl.java:409)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.a(FileDocumentManagerImpl.java:383)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveDocument(FileDocumentManagerImpl.java:334)
    at com.intellij.openapi.fileEditor.impl.FileDocumentManagerImpl.saveDocument(FileDocumentManagerImpl.java:326)
    at com.magicento.helpers.IdeHelper$2.run(IdeHelper.java:436)
    at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
    at com.magicento.helpers.IdeHelper$Reformatter$1.run(IdeHelper.java:467)
    at com.intellij.openapi.application.impl.ApplicationImpl.invokeAndWait(ApplicationImpl.java:711)
    at com.magicento.helpers.IdeHelper$Reformatter.execute(IdeHelper.java:464)
    at com.magicento.helpers.IdeHelper.reformat(IdeHelper.java:442)
    at com.magicento.extensions.MagicentoXmlCompletionContributor$1.handleInsert(MagicentoXmlCompletionContributor.java:112)
    at com.intellij.codeInsight.lookup.LookupElementBuilder.handleInsert(LookupElementBuilder.java:369)
    at com.intellij.codeInsight.lookup.LookupElementDecorator.handleInsert(LookupElementDecorator.java:64)
    at com.intellij.codeInsight.completion.CodeCompletionHandlerBase$10.run(CodeCompletionHandlerBase.java:759)
    at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:1010)
    at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.a(CodeCompletionHandlerBase.java:749)
    at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.access$400(CodeCompletionHandlerBase.java:72)
    at com.intellij.codeInsight.completion.CodeCompletionHandlerBase$7.perform(CodeCompletionHandlerBase.java:658)
    at com.intellij.openapi.editor.impl.CaretModelImpl$3.run(CaretModelImpl.java:355)
    at com.intellij.openapi.editor.impl.CaretModelImpl.doWithCaretMerging(CaretModelImpl.java:435)
    at com.intellij.openapi.editor.impl.CaretModelImpl.runForEachCaret(CaretModelImpl.java:346)
    at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.a(CodeCompletionHandlerBase.java:651)
    at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lookupItemSelected(CodeCompletionHandlerBase.java:590)
    at com.intellij.codeInsight.completion.CompletionProgressIndicator$2.itemSelected(CompletionProgressIndicator.java:115)
    at com.intellij.codeInsight.lookup.impl.LookupImpl.fireItemSelected(LookupImpl.java:854)
    at com.intellij.codeInsight.lookup.impl.LookupImpl.finishLookup(LookupImpl.java:526)
    at com.intellij.codeInsight.lookup.impl.LookupImpl.finishLookup(LookupImpl.java:465)
    at com.intellij.codeInsight.lookup.impl.actions.ChooseItemAction$Handler.execute(ChooseItemAction.java:71)
    at com.intellij.openapi.editor.actionSystem.EditorActionHandler.doExecute(EditorActionHandler.java:170)
    at com.intellij.openapi.editor.actionSystem.EditorActionHandler$4.perform(EditorActionHandler.java:215)
    at com.intellij.openapi.editor.actionSystem.EditorActionHandler.doIfEnabled(EditorActionHandler.java:107)
    at com.intellij.openapi.editor.actionSystem.EditorActionHandler.execute(EditorActionHandler.java:212)
    at com.intellij.openapi.editor.actionSystem.EditorAction$1.run(EditorAction.java:97)
    at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:124)
    at com.intellij.openapi.editor.actionSystem.EditorAction.actionPerformed(EditorAction.java:108)
    at com.intellij.openapi.editor.actionSystem.EditorAction.actionPerformed(EditorAction.java:82)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher$3.performAction(IdeKeyEventDispatcher.java:597)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.processAction(IdeKeyEventDispatcher.java:646)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.d(IdeKeyEventDispatcher.java:496)
    at com.intellij.openapi.keymap.impl.IdeKeyEventDispatcher.dispatchKeyEvent(IdeKeyEventDispatcher.java:206)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:546)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:384)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: java.lang.IllegalStateException: Current caret is defined, cannot operate on other ones
    at com.intellij.openapi.editor.impl.CaretModelImpl.runForEachCaret(CaretModelImpl.java:344)
    at com.intellij.openapi.editor.impl.CaretModelImpl.runForEachCaret(CaretModelImpl.java:333)
    at com.intellij.openapi.editor.impl.TrailingSpacesStripper.stripIfNotCurrentLine(TrailingSpacesStripper.java:182)
    at com.intellij.openapi.editor.impl.TrailingSpacesStripper.a(TrailingSpacesStripper.java:81)
    at com.intellij.openapi.editor.impl.TrailingSpacesStripper.beforeDocumentSaving(TrailingSpacesStripper.java:61)
    ... 57 more
```

Magicento version: 2.8.6
PhpStorm version: PS-138.1901

**Comment by toddb02** [comment]
Also getting this often after upgrade to 2.8.6 on PHPStorm 8.0.1, Build PS-138.2001

**Comment by dsitovs** [comment]
Hello,

sorry, getting the same issue guys ;)

Magicento 2.8.6,
Linux Mint 17 x64, PhpStorm 8.0.1, Build PS-138.2001.

**Comment by ndlinh** [comment]
Same issue.

Magicento 2.8.6, PhpStorm 8.0.1, OSX 10.10

**Comment by therouv** [comment]
Same issue with Magicento 2.8.6 and PhpStorm 8.0.1 on OSX 10.10

**Comment by enriquepiatti** [comment]
Should be fixed since 2.9+ please reopen if not the case
